### PR TITLE
Wallet plugin fixes

### PR DESCRIPTION
**Pull request by pmknutsen**
Fixed broken Address search
More intuitive page navigation
Enforce limits on requested page
Compare _absolute_ transaction values in transaction filter
Add a reasonable minimum value to avoid displaying dust
Increase font size in Wallet to 12px
Darker red and green on transaction icons
Improved formatting of small Siacoin amounts (show n'th significant digits vs 2 decimals for amounts > 1 SC)

screenshot from 2016-01-24 22 30 00

### CODE REVIEW

File: plugins/Wallet/js/transactions.js [hunk 1]
```diff
@@ -33,8 +33,8 @@ function checkCriteria(txn) {
 	// Failing one check returns false but passing all checks returns true
 	return (criteria.endTime   === null || txn.confirmationtimestamp <= criteria.endTime)   &&
 	       (criteria.startTime === null || txn.confirmationtimestamp >= criteria.startTime) &&
-	       (criteria.maxValue  === null || txn.value                 <= criteria.maxValue)  &&
-	       (criteria.minValue  === null || txn.value                 >= criteria.minValue)  &&
+	       (criteria.maxValue  === null || Math.abs(txn.value)       <= criteria.maxValue)  &&
+	       (criteria.minValue  === null || Math.abs(txn.value)       >= criteria.minValue)  &&
```
Comment (Mingling94) [inline plugins/Wallet/js/transactions.js:37]: Great catch!

File: plugins/Wallet/js/transactions.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ function makeTransaction(txn) {
 	element.find('.time').html(time);
 
 	// Set transaction positive or negative
-	if (txn.value.isNegative()) {
+	if (txn.value < 0) {
```
Comment (Mingling94) [inline plugins/Wallet/js/transactions.js:93]: Does this work? I thought from line 68, `txn.value` is a `BigNumber` object?

Comment (pmknutsen) [inline plugins/Wallet/js/transactions.js:93]: It works. For some reason I don't understand, when you call toPrecision() on the BigNum it seems to be cast to a real number and isNegative() is no longer an available method. I tested it on negative and positive cases.

**Comment by DavidVorick** [comment]
Looks good to me.

File: plugins/Wallet/js/helpers.js [hunk 3]
```diff
@@ -12,10 +12,12 @@ function convertSiacoin(hastings) {
 	// This deals with that imperfectly
 	var number = new BigNumber(hastings);
 	var ConversionFactor = new BigNumber(10).pow(24);
-	return number.dividedBy(ConversionFactor).round(2);
+
+	return number.gt(1) ? number.dividedBy(ConversionFactor).round(2)
+						 : number.dividedBy(ConversionFactor).toPrecision(1);
```
Comment (Mingling94) [inline plugins/Wallet/js/helpers.js]: Ah then this is the problem. `round()` returns a `BigNumber` and `toPrecision()` returns a `string`. This function should return one type of value.

File: plugins/Wallet/js/transactions.js (hunk 1 above)
Comment (Mingling94) [inline plugins/Wallet/js/transactions.js:37]: These lines also depend on what type of value is returned from `convertSiacoin()` and set per transaction object.

**Pull request merged**